config/components/certmanager/certificate-metrics.yaml

```diff
@@ -7,11 +7,11 @@ metadata:
   namespace: system
 spec:
   dnsNames:
```

Comment:
There is a chance curl didn't work for you because we didn't include a subject (e.g. `commonName`) on our certificates. There is an issue that some libraries might not recognize such self-signed cert as valid. 

Please see https://cert-manager.io/docs/configuration/selfsigned/#certificate-validity for more details. 

However, I could not reproduce that in my pod.

Reply:
Yeah, I think `commonName` played the trick here.